src/Features/CSharp/Portable/DocumentationComments/DocumentationCommentSnippetService.cs

```diff
@@ -111,22 +124,106 @@ protected override List<string> GetDocumentationCommentStubLines(MemberDeclarati
                 }
             }
 
-            if (member.IsKind(SyntaxKind.MethodDeclaration) ||
-                member.IsKind(SyntaxKind.IndexerDeclaration) ||
-                member.IsKind(SyntaxKind.DelegateDeclaration) ||
-                member.IsKind(SyntaxKind.OperatorDeclaration))
+            if (member.IsKind(
+                    SyntaxKind.MethodDeclaration,
+                    SyntaxKind.IndexerDeclaration,
+                    SyntaxKind.DelegateDeclaration,
+                    SyntaxKind.OperatorDeclaration,
+                    SyntaxKind.ConstructorDeclaration,
+                    SyntaxKind.DestructorDeclaration))
             {
                 var returnType = member.GetMemberType();
                 if (returnType != null &&
                     !(returnType.IsKind(SyntaxKind.PredefinedType, out PredefinedTypeSyntax? predefinedType) && predefinedType.Keyword.IsKindOrHasMatchingText(SyntaxKind.VoidKeyword)))
                 {
                     list.Add("/// <returns></returns>");
                 }
+
+                foreach (var exceptionType in GetExceptions(member, model, options, cancellationToken))
+                {
+                    list.Add(@$"/// <exception cref=""{exceptionType}""></exception>");
+                }
             }
 
             return list;
         }
 
+        private static IEnumerable<string> GetExceptions(SyntaxNode member, SemanticModel model, OptionSet options, CancellationToken cancellationToken)
+        {
+            var throwExpressionsAndStatements = member.DescendantNodes().Where(n => n.IsKind(SyntaxKind.ThrowExpression, SyntaxKind.ThrowStatement));
```

Comment:
this will find in nested lambdas/localfunctions

Reply:
Finding in nested local functions (possible lambdas too) is okay since they're expected to be called within the method. However, a `try catch` wrapping the local function call seems tricky to handle. Is this an important scenario?

Reply:
i could see the argument about nested local functions (and even the it's iffy).  not lambdas though.   I would personally not handle either unless you can actually see the local function being called (as opposed to passed as a delegate).

Reply:
and i do think we should basically be doing this analysis as finding the locations in the method where exceptions can be thrown, but then filtering those out if they are contained ina  try/catch that would catch that particular exception.

Reply:
this comment still applies.